streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -390,6 +396,40 @@ public void shouldNotOverrideUserConfigCommitIntervalMsIfExactlyOnceEnabled() {
         assertThat(streamsConfig.getLong(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG), equalTo(commitIntervalMs));
     }
 
+    @Test
+    public void shouldBeBackwardsCompatibleWithDeprecatedConfigs() {
+        final Properties props = minimalStreamsConfig();
+        props.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serdes.Double().getClass());
+        props.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.Double().getClass());
+        props.put(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, MockTimestampExtractor.class);
+
+        final StreamsConfig config = new StreamsConfig(props);
+        MatcherAssert.assertThat(config.defaultKeySerde() instanceof Serdes.DoubleSerde, is(true));
```

Comment:
Why not a simple `assertTrue(config.defaultKeySerde() instanceof Serdes.DoubleSerde)` ?